source/fundamentals/crud/write-operations/upsert.txt

```diff
@@ -4,8 +4,105 @@ Insert or Update in a Single Operation
 
 .. default-domain:: mongodb
 
-If your application stores and modifies data in MongoDB, you probably use
-insert and update operations. In certain workflows, you may need to choose
-between an insert and update depending on whether the document exists.
-In these cases, you can streamline your application logic by using the
-``upsert`` option available in the following methods:
+Overview
+--------
+
+You often use insert and update operations for applications that store
+and modify data. Sometimes, you need to choose between an insert or
+update depending on whether the document exists. 
+
+To insert or update documents, pass a query filter to the 
+:java-docs:`updateOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateOne(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`, 
+:java-docs:`updateMany() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#updateMany(org.bson.conversions.Bson,org.bson.conversions.Bson,com.mongodb.client.model.UpdateOptions)>`
+or :java-docs:`replaceOne() </apidocs/mongodb-driver-sync/com/mongodb/client/MongoCollection.html#replaceOne(org.bson.conversions.Bson,TDocument,com.mongodb.client.model.ReplaceOptions)>`
+methods. 
+
+These methods update documents that match your query filter,
+but inserts a document if the there are no matches when you set the
+upsert to true. When multiple documents match the query filter supplied to
+``updateOne()`` or ``replaceOne()``, the first document that matches
+gets updated.
```

Comment:
**issue:** This does not get to the point of _this_ page. In reading this, you should move this content to a separate page for **Update** operations, then get into Upsert.

To introduce Upsert in the introduction, I would recommend something closer to this:
```suggestion
Applications don't only store data. Data changes over time. You can use Update operations
to modify existing data. Update operations assume the data that you want to change exists.
This creates a wrinkle when developing your application. You would write logic to check for
the data first. MongoDB offers an easier option called **Upsert**.
```